src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const array = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of action.keysToRemove) {
+          delete newState[i];
+        };
+        break;
+
+      case 'clear':
+        Object.keys(newState).forEach(key => delete newState[key]);
```

Comment:
`switch / case` should have `default` statement. You can handle some errors inside it, for example